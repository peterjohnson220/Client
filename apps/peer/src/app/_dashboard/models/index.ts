export * from './existing-company.model';
