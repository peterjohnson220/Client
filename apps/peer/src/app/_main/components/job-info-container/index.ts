export * from './job-info-container.component';
