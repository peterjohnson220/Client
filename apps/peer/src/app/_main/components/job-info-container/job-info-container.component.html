<div class="info-container">
  <div class="info-row">
    <div class="label">{{job.JobType}} Job:</div>
    <div>{{ job.JobTitle }} <ng-container *ngIf="showJobCode"> - {{job.JobCode}}</ng-container></div>
  </div>
  <div class="info-row">
    <div class="label">Job Family:</div>
    <div>{{ job.JobFamily }}</div>
  </div>
  <div class="info-row">
    <div class="label">Job Level:</div>
    <div>{{ job.JobLevel }}</div>
  </div>
  <div class="info-row-flex-column flex-grow-1">
    <div class="label">Job Description:</div>
    <div class="job-description-container">
      {{ job.JobDescription }}
    </div>
  </div>
</div>
