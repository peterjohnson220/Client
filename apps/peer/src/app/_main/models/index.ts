export * from './existing-company.model';
export * from './payfactors-job.model';
