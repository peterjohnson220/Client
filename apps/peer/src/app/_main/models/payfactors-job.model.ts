export interface PayfactorsJob {
  MDJobsBaseId: number;
  JobTitle: string;
  JobFamily: string;
  JobLevel: string;
  JobDescription: string;
  InExchange: boolean;
  PendingRequest: boolean;
}

export function generateMockPayfactorsJob(): PayfactorsJob {
  return {
    MDJobsBaseId: 0,
    JobTitle: 'MockJobTitle',
    JobFamily: 'MockJobFamily',
    JobLevel: 'MockJobLevel',
    JobDescription: 'MockJobDescription',
    InExchange: false,
    PendingRequest: false
  };
}
