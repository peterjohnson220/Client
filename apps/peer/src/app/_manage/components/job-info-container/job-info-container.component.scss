@import 'colors';

:host{
  flex: 0 0 50%;
}

.info-container{
  display: flex;
  flex-direction: column;
  padding-left: 1rem;
  padding-right: 1rem;
  height: 100%;
}

.job-description-container{
  padding-right: 1rem;
  margin-right: -1rem;
  margin-top: .5rem;
  flex-grow: 1;
  overflow-y: auto;
  line-height: 1.7;
}

%info-row-shared{
  display: flex;
  padding: .65rem .25rem;
  border-bottom: 1px solid $border-grey;

  .label{
    font-weight: bold;
    width: 25%;
  }
}

.info-row{
  @extend %info-row-shared;
  min-height: 40px;
}

.info-row-flex-column{
  @extend %info-row-shared;
  flex-direction: column;
  border-bottom: none;
}
