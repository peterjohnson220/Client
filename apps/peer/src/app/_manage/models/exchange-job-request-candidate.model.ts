export interface ExchangeJobRequestCandidate {
  MDJobsBaseId: number;
  JobTitle: string;
  JobFamily: string;
  JobLevel: string;
  JobDescription: string;
  InExchange: boolean;
  PendingRequest: boolean;
}

export function generateMockExchangeJobRequestCandidate(): ExchangeJobRequestCandidate {
  return {
    MDJobsBaseId: 0,
    JobTitle: 'MockJobTitle',
    JobFamily: 'MockJobFamily',
    JobLevel: 'MockJobLevel',
    JobDescription: 'MockJobDescription',
    InExchange: false,
    PendingRequest: false
  };
}
