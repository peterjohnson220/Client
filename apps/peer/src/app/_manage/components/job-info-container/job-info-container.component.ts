import { Component, Input } from '@angular/core';

import { Job } from 'libs/models';

@Component({
  selector: 'pf-job-info-container',
  templateUrl: './job-info-container.component.html',
  styleUrls: ['./job-info-container.component.scss']
})
export class JobInfoContainerComponent {
  @Input() job: Job;
  @Input() showJobCode: boolean;

  constructor() {}
}
