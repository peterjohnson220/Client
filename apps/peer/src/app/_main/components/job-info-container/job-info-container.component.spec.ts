import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { generateMockJob } from 'libs/models';

import { JobInfoContainerComponent } from './job-info-container.component';

describe('Peer - Job Info Container', () => {
  let fixture: ComponentFixture<JobInfoContainerComponent>;
  let instance: JobInfoContainerComponent;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        JobInfoContainerComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(JobInfoContainerComponent);
    instance = fixture.componentInstance;
  });

  it('should show the job code, when showJobCode is true', () => {
    instance.job = generateMockJob();
    instance.showJobCode = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

});
