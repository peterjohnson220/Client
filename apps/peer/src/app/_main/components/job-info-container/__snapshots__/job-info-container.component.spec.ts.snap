// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peer - Job Info Container should show the job code, when showJobCode is true 1`] = `
<pf-job-info-container
  job={[Function Object]}
  showJobCode={[Function Boolean]}
>
  <div
    class="info-container"
  >
    
    
    <div
      class="info-row"
    >
      
      
      <div
        class="label"
      >
        Company Job:
      </div>
      
      
      <div>
        Mock Job 
        
        
         - 1002
      </div>
      
    
    </div>
    
    
    <div
      class="info-row"
    >
      
      
      <div
        class="label"
      >
        Job Family:
      </div>
      
      
      <div>
        Mock Family
      </div>
      
    
    </div>
    
    
    <div
      class="info-row"
    >
      
      
      <div
        class="label"
      >
        Job Level:
      </div>
      
      
      <div>
        Mock Level
      </div>
      
    
    </div>
    
    
    <div
      class="info-row-flex-column flex-grow-1"
    >
      
      
      <div
        class="label"
      >
        Job Description:
      </div>
      
      
      <div
        class="job-description-container"
      >
        
        Lorem Ipsum
      
      </div>
      
    
    </div>
    
  
  </div>
  
</pf-job-info-container>
`;
